Source/ACE.Server/Managers/WorldManager.cs

```diff
@@ -78,9 +81,55 @@ public static void Initialize()
             log.DebugFormat($"Current maximum allowed sessions: {ConfigManager.Config.Server.Network.MaximumAllowedSessions}");
         }
 
-        public static void ProcessPacket(ClientPacket packet, IPEndPoint endPoint)
+        public static void ProcessPacket(ClientPacket packet, IPEndPoint endPoint, IPEndPoint listenerEndpoint)
         {
-            if (packet.Header.HasFlag(PacketHeaderFlags.LoginRequest))
+            // TO-DO: generate ban entries here based on packet rates of endPoint, IP Address, and IP Address Range
+            if (listenerEndpoint.Port == ConfigManager.Config.Server.Network.Port + 1)
+            {
+                if (packet.Header.Flags.HasFlag(PacketHeaderFlags.ConnectResponse))
+                {
+                    PacketInboundConnectResponse connectResponse = new PacketInboundConnectResponse(packet);
+
+                    // This should be set on the second packet to the server from the client.
+                    // This completes the three-way handshake.
+                    // the client should be sending it back now
+                    // This could be offloaded into a queue and processed on another thread
+                    // TO-DO: generate ban entries here, situation rate
+                    // TO-DO: fix race condition between bob and eve getting and setting session state
+                    var session =
```

Comment:
when iterating over the sessionMap, we must enter a:
sessionLock.EnterReadLock();

Alternatively you can create a private Find(args...) function below to find the specific session you're looking for, following the pattern the other Find() functions use.